llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -197,6 +198,26 @@ static bool expandLog10Intrinsic(CallInst *Orig) {
   return expandLogIntrinsic(Orig, numbers::ln2f / numbers::ln10f);
 }
 
+static bool expandPowIntrinsic(CallInst *Orig) {
```

Comment:
It's strange how the function has no choice but to return true. I see other similar functions do the same. Would it break convention to instead have this be a void function, and directly return true at the call site? I think it would be less confusing.

Reply:
Or, perhaps false can be returned in certain cases, like if Orig is null or if Ty happens to be invalid.

Reply:
Its a good question the reason why is  `expandIntrinsic` I need to be able to return true for the intrinsics that are expanded so if we made these void returns then
```c++
switch (F.getIntrinsicID()) {
  case Intrinsic::abs:
    expandAbs(Orig);
   return true;
  ...
  case Intrinsic::pow:
    expandPowIntrinsic(Orig);
    return true;
```
Having all those return trues in the switch statement was just ugly so i moved them into the function bodies.

Reply:
Agreed, in terms of aesthetics, it would look bad. However, (this may not be the PR to make this change), maybe this could be considered:

switch (F.getIntrinsicID()) {
  case <a pre-existing case>:
     expand<whatever function>(...);
     break;
  ....
  default:
     return false;
  }
  return true;

There would be break statements in each case, but that could be an improvement? This is a bit pedantic but it feels slightly icky to have a function return a boolean unconditionally.

Reply:
It really should be a return as that will break the flow of execution and end the function.  ending the function execution as soon as possible is a defensive coding strategy we should pursue.  I also don't find break anymore aesthetically pleasing than `return true;` both would be adding a second line per case statement.

Reply:
> Or, perhaps false can be returned in certain cases, like if Orig is null or if Ty happens to be invalid.

I think for invalid cases we probably want to end execution something like `llvm_unreachable` having a return false in this case wouldn't make us handle the invalid case it would just prevent further removal of the function we are expanding.

Reply:
@bob80905  If you feel strongly about this you or I could put up a pr converting the expansion functions to  void and returning true in the switch statements.  I don't think aesthetics should trump understanding of intent of the bools.